docs/sources/flow/getting-started/controllers.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Kuberenetes Deployment Modes
+---
+## StatefulSet
+
+A `StatefulSet` is the most common way Grafana Agent is deployed in Kubernetes. Advantages include:
+
+- A relatively small number of Agents can handle metrics for an entire cluster.
+- Persistent volumes can be attatched to each pod to persist agent data like the write-ahead log or other component state.
+- Number of pods can be scaled up or down as needed, either manually or with horizontal pod autoscaling.
+
+### Distributing work with StatefulSets
+
+Grafana Agent's [built-in clustering](https://grafana.com/docs/agent/latest/flow/getting-started/distribute-prometheus-scrape-load/) is the easiest way to distribute scrape load across a `StatefulSet` of Agents. 
+
+## Deployment
+
+A `Deployment` has most of the same properties as a `StatefulSet`, but cannot use Persistent Volume Claim Templates. It may be desired if you do not want storage, but most of the other properties are the same as a `StatefulSet`.
+
+## DaemonSet
+
+A `DaemonSet` will deploy a Grafana Agent pod to each Node in your Kubernetes cluster. This is needed when you need access to something that is present on every Node in your cluster, such as:
+
+- Local access to log files.
+- Exporters, such as `node_exporter`, which run on every physical node.
+
+There are several downsides to a `DaemonSet` which need to be considered:
+
+- You cannot attach Persistent Volumes to `DaemonSet` pods. This means things like the Write-Ahead Log will use the Node's local disk, which may not be desired in shared environments.
```

Comment:
Depending on the audience's expected familiarity, it might be good to note an example of why it wouldn't be desired (e.g. noisy neighbor; if any other pods on the node really need the disk there could be contention, slow, etc...)?